modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorRuleHelper.java

```diff
@@ -151,8 +158,42 @@ protected boolean containsAction(
 			action -> Objects.equals(setPropertyAction, action));
 	}
 
+	protected boolean containsCalculatedAction(
```

Comment:
In this function, I'm evaluating more complex expressions affected by a calculate rule. The purpose here is, for each rule, verify if the field can get the correct value after gets out of some action. As the field only gets here if they are not affected by the rule anymore and the _defaultValue_ is the  previous value defined by the rule, I'm checking if the current value is the same as if I had to evaluate the field. The call for the method calculateAction returns all the fields evaluated by a specific rule, this was necessary for compare both values. The conditional in line 171 was created to minimize the number of time that fields goes into this calls and do the evaluation. In the end (182), there are two verifications: one if the values are the same as I said before, but also if the result of the fields are 0 (the use case I tried was when fields are empty, the affected field should return to the predefined or empty value).